app/controllers/works_controller.rb

```diff
@@ -0,0 +1,89 @@
+class WorksController < ApplicationController
+
+  def index
+    @works = Work.all
+  end
+
+  def show
+    @work = Work.find_by(id: params[:id])
+
+    if @work.nil?
+      head :not_found
+      return
```

Comment:
In the future maybe just do redirects with a flash notice.  Much better for end-users.